### PR TITLE
Integrate yeoman generators skeleton with oclif commands - Closes #6067

### LINKED ISSUE #6067: Integrate yeoman generators skeleton with oclif commands
### Description

We need to use `yeoman` as bootstrapping tool but encapsulated in `oclif` CLI interface. So the yeoman generators will be part of the `commander` and only accessible within `commander` cli. 

### Motivation

With this encapsulated approach, we will be able to reuse all `yeoman` generators from inside `commander` context. This way the `commander` can be used a utility CLI as well as bootstrapping tool. 

### Acceptance Criteria

- Only the `oclif` CLI to be exposed to user 
- The generators should not be exported from the commander

### Additional Information

- For this integration we can't use `yo` but rather `yeoman-environment`

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #6067

### How was it solved?

- Implemented the integration of `oclif` and `yeoman` so they can work together from one interface. 

### How was it tested?

- No tests added in this PR, more relevant tests will come in other issues.

### CODE REVIEW

File: commander/src/base_bootstrap_command.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2019 Lisk Foundation
```
Comment (ManuGowda) [inline commander/src/base_bootstrap_command.ts:3]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```

File: commander/src/commands/init.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2020 Lisk Foundation
```
Comment (ManuGowda) [inline commander/src/commands/init.ts:3]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```

File: commander/src/commands/init.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+	async run(): Promise<void> {
+		return new Promise((resolve, reject) => {
+			env.run(
+				'lisk:init',
+				// TODO: Pass the commander version
+				{ template: this.bootstrapFlags.template, version: '0.1.0' },
+				(err): void => {
+					if (err) {
+						console.error(err);
```
Comment (ManuGowda) [inline commander/src/commands/init.ts:33]:
```suggestion
						this.error(err);
```

File: commander/src/generators/base_generator.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2020 Lisk Foundation
```
Comment (ManuGowda) [inline commander/src/generators/base_generator.ts:3]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```

File: commander/src/generators/base_generator.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ */
+
+import Generator from 'yeoman-generator';
+import Storage from 'yeoman-generator/lib/util/storage';
+import { join, dirname } from 'path';
+import { BootstrapGeneratorOptions } from '../types';
+
+// const DEFAULT_TEMPLATE_NAME = 'lisk-ts';
+
```
Comment (ManuGowda) [inline commander/src/generators/base_generator.ts:23]: if not used, then we can delete?

File: commander/src/generators/base_generator.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import Generator from 'yeoman-generator';
```
Comment (ManuGowda) [inline commander/src/generators/base_generator.ts:17]:
```suggestion
import YeomanGenerator from 'yeoman-generator';
```

File: commander/src/generators/base_generator.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import Storage from 'yeoman-generator/lib/util/storage';
+import { join, dirname } from 'path';
+import { BootstrapGeneratorOptions } from '../types';
+
+// const DEFAULT_TEMPLATE_NAME = 'lisk-ts';
+
+export default abstract class BaseGenerator extends Generator {
```
Comment (ManuGowda) [inline commander/src/generators/base_generator.ts:24]:
```suggestion
export default abstract class BaseGenerator extends YeomanGenerator {
```

File: commander/src/generators/env.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2020 Lisk Foundation
```
Comment (ManuGowda) [inline commander/src/generators/env.ts:3]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```

File: commander/src/generators/init.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
+/*
+ * LiskHQ/lisk-commander
+ * Copyright © 2020 Lisk Foundation
```
Comment (ManuGowda) [inline commander/src/generators/init.ts:3]:
```suggestion
 * Copyright © 2021 Lisk Foundation
```

File: commander/src/generators/init.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+export default class InitGenerator extends BaseGenerator {
+	private answers!: InitPrompts;
+
+	async prompting() {
+		this.answers = (await this.prompt([
+			{
+				type: 'input',
+				name: 'name',
+				message: 'Your app name',
```
Comment (ManuGowda) [inline commander/src/generators/init.ts:30]:
```suggestion
				message: 'Application name',
```

File: commander/src/generators/init.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+				default: basename(this.destinationRoot()),
+			},
+			{
+				type: 'input',
+				name: 'description',
+				message: 'Your app description',
```
Comment (ManuGowda) [inline commander/src/generators/init.ts:36]:
```suggestion
				message: 'Application description',
```

**Review by ManuGowda — COMMENTED** [review]
LGTM, can we add `.nvmrc` to the template?

File: commander/src/commands/init.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+	static flags = {
+		...BaseBootstrapCommand.flags,
+	};
+
+	async run(): Promise<void> {
+		return new Promise((resolve, reject) => {
+			env.run(
+				'lisk:init',
+				// TODO: Pass the commander version
+				{ template: this.bootstrapFlags.template, version: '0.1.0' },
```
Comment (shuse2) [inline commander/src/commands/init.ts:30]: `this.config` contains package.json

File: commander/src/base_bootstrap_command.ts [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ */
+import { Command, flags as flagParser } from '@oclif/command';
+
+interface BootstrapFlags {
+	readonly template: string;
+}
+
+export default abstract class BaseBootstrapCommand extends Command {
+	static flags = {
```
Comment (shuse2) [inline commander/src/base_bootstrap_command.ts:23]: if it's just a flag, does it needs to be base command? I think it's ok to just use the common flag variable

**Review by shuse2 — COMMENTED** [review]
LGTM in general

File: commander/src/base_bootstrap_command.ts (hunk 13 above)
Comment (nazarhussain) [reply commander/src/base_bootstrap_command.ts:23]: Along with flag there is parsing logic and some other code will come which is useful for all bootstrpping commands so I think its good to keep it this way. We also used similar approach on other places.

File: commander/src/commands/init.ts (hunk 12 above)
Comment (nazarhussain) [reply commander/src/commands/init.ts:30]: This update is coming in upcoming PR.

File: commander/src/generators/base_generator.ts (hunk 5 above)
Comment (nazarhussain) [reply commander/src/generators/base_generator.ts:23]: Its uncommented and used in the second PR.

**Review by ManuGowda — APPROVED** [review]

**Review by shuse2 — APPROVED** [review]

**Pull request merged**